### DISCUSSION TITLE
v1.119.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#13778] [Announcements]
# v1.119.0

> [!CAUTION]
> 
> The env variable for the host binding was erroneously named `HOST` instead of `IMMICH_HOST` (which is how it was listed in the docs). This has been corrected in this release.
> If you were using the `HOST` env var in your setup before, please update it to `IMMICH_HOST`.
> 
> **If you are using the built-in Prometheus endpoint for monitoring, please read on. If not, you can ignore this section.**
> 
> The following env variables have been removed:
> - `IMMICH_METRICS`
> - `IMMICH_API_METRICS`
> - `IMMICH_HOST_METRICS`
> - `IMMICH_IO_METRICS`
> - `IMMICH_JOB_METRICS`
> 
> Use `IMMICH_TELEMETRY_INCLUDE` / `IMMICH_TELEMETRY_EXCLUDE` instead.
>
> ### Examples:
>
> ```diff
> -- IMMICH_METRICS=true
> ++ IMMICH_TELEMETRY_INCLUDE=all
> ```
>
> ```diff
> -- IMMICH_METRICS=true
> -- IMMICH_HOST_METRICS=false
> ++ IMMICH_TELEMETRY_INCLUDE=all
> ++ IMMICH_TELEMETRY_EXCLUDE=host
> ```
>
> ```diff
> -- IMMICH_API_METRICS=true
> -- IMMICH_HOST_METRICS=true
>++ IMMICH_TELEMETRY_INCLUDE=api,host
> ```

Welcome to release `v1.119.0` of Immich. This release focuses on fixing issues surrounding video encoding and thumbnail generation for a variety of old and new formats, along with minor improvements across the app. Some of the highlights below:

- Create new OAuth users without passwords
- Debounce email notifications for album changes
- Improve mobile app layout on tablets
- VAAPI hardware decoding
- Notable fix: Face Detection using excessive RAM with OpenVINO
- Notable fix: transcodes not playing in the mobile app in certain situations

### VAAPI hardware decoding

Prior to this release, Immich only used VAAPI for video encoding; it still used the CPU for decoding and tone-mapping. This release adds end-to-end acceleration for VAAPI, making transcoding much faster with lower CPU utilization. With this addition, every supported acceleration API now has end-to-end acceleration in Immich.

If you use VAAPI hardware acceleration, you can enable hardware decoding in the [transcoding settings](https://my.immich.app/admin/system-settings?isOpen=video-transcoding+hardware-acceleration) to take advantage of this change.

### Transcoding fix for mobile

When the source video is Dolby Vision, transcodes sometimes either didn't play in the mobile app, or only the audio was played. In particular, this meant that many videos captured on iOS could not be viewed on Android.

This issue has been fixed. However, existing transcodes of Dolby Vision videos created using Immich 1.114.0 - 1.118.2 will still have the issue and will need to be re-transcoded for correct playback in these cases.

To re-transcode affected videos, you can either select them in the web app and choose the *Refresh encoded videos* option or go to the [Job Status page](http://my.immich.app/admin/jobs-status) and click *All* next to Transcode Videos.

Note: you only need to do this if you encounter playback issues for these videos.

## Support Immich

<p align="center">
<img src="https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExbjY2eWc5Y2F0ZW56MmR4aWE0dDhzZXlidXRmYWZyajl1bWZidXZpcyZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/87CKDqErVfMqY/giphy.gif" width="450" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich by purchasing a product key at [https://buy.immich.app](https://buy.immich.app).

Cheers! 🍻

- - - -

### 🚨 Breaking Changes
* refactor(server): move host env to config repo by @jrasm91 in https://github.com/immich-app/immich/pull/13507
* refactor(server): telemetry env variables by @danieldietzler in https://github.com/immich-app/immich/pull/13705
### 🚀 Features
* feat(server): vaapi hardware decoding by @mertalev in https://github.com/immich-app/immich/pull/13561
### 🌟 Enhancements
* feat: show warning when running main branch build by @bo0tzz in https://github.com/immich-app/immich/pull/13462
* fix(server): Allow passwordless users when oauth enabled by @jedi04 in https://github.com/immich-app/immich/pull/13517
* feat(web): Change relink person icon from minus to pencil by @CrushedAsian255 in https://github.com/immich-app/immich/pull/13536
* fix(web): improve photosphere viewer settings by @grgergo1 in https://github.com/immich-app/immich/pull/13468
* feat(server): add pcm_s16le accepted audio codec by @pyorot in https://github.com/immich-app/immich/pull/13418
* feat(server): wait five minutes before sending email on new album item by @HeyBanditoz in https://github.com/immich-app/immich/pull/12223
* chore(mobile): proper new UI layout for tablet by @alextran1502 in https://github.com/immich-app/immich/pull/13650
* chore(mobile): thumbnail image ui improvements  by @dvbthien in https://github.com/immich-app/immich/pull/13655
* feat: Use video thumbnail for GIF types by @Pranay-Pandey in https://github.com/immich-app/immich/pull/13677
* feat(ml): configurable batch size for facial recognition by @mertalev in https://github.com/immich-app/immich/pull/13689
### 🐛 Bug fixes
* fix(server): never try to parse Duration from exif data by @jrasm91 in https://github.com/immich-app/immich/pull/13497
* fix: person update state inconsistencies by @danieldietzler in https://github.com/immich-app/immich/pull/13556
* fix: web search add to album reactivity by @danieldietzler in https://github.com/immich-app/immich/pull/13539
* fix(web): intersection observer not triggered to load more people by @alextran1502 in https://github.com/immich-app/immich/pull/13589
* fix(server): encodes iPhone 16 Pro video with unknown audio codec by @alextran1502 in https://github.com/immich-app/immich/pull/13593
* fix(server): Revert "fix(server): copy video projection metadata for 360 videos (#12376)" by @mertalev in https://github.com/immich-app/immich/pull/13611
* fix(mobile): Fix minor issues with downloading assets by @dvbthien in https://github.com/immich-app/immich/pull/13609
* fix(web): reset select all button state on escape press by @darrelhong in https://github.com/immich-app/immich/pull/13600
* fix(server): only allow absolute import paths by @etnoy in https://github.com/immich-app/immich/pull/13642
* feat(web): allow changing shared user role in album options modal by @AnimeshKotka in https://github.com/immich-app/immich/pull/13654
* fix(server): add ld_library_path in start.sh by @mertalev in https://github.com/immich-app/immich/pull/13686
* fix(server): nan transcoding estimate by @mertalev in https://github.com/immich-app/immich/pull/13693
* fix: Add debouncing to the search popup form by @Pranay-Pandey in https://github.com/immich-app/immich/pull/13684
* fix(web): add crossorigin to link manifest for PWA by @automaton82 in https://github.com/immich-app/immich/pull/13708
* fix: duplicated library scan jobs and api server library watch by @zackpollard in https://github.com/immich-app/immich/pull/13734
* fix: X years ago ordering by @darrelhong in https://github.com/immich-app/immich/pull/13742
### 📚 Documentation
* chore(docs): add Thai README by @richeyphu in https://github.com/immich-app/immich/pull/13591
* chore(docs): update _storage-template.md by @chriskoch in https://github.com/immich-app/immich/pull/13578
* docs: add qsv note for jasper lake cpus by @evie-lau in https://github.com/immich-app/immich/pull/13622
* docs: add warning to repair page by @polgarc in https://github.com/immich-app/immich/pull/13640
* feat(docs): add mention to purchase to support immich page by @CrushedAsian255 in https://github.com/immich-app/immich/pull/13612
* fix: Add DB_DATA_LOCATION config. by @Sweebah in https://github.com/immich-app/immich/pull/13754
### 🌐 Translations
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/13443
* chore: move weblate translations to root directory by @ben-basten in https://github.com/immich-app/immich/pull/13604

## New Contributors
* @jedi04 made their first contribution in https://github.com/immich-app/immich/pull/13517
* @CrushedAsian255 made their first contribution in https://github.com/immich-app/immich/pull/13536
* @grgergo1 made their first contribution in https://github.com/immich-app/immich/pull/13468
* @richeyphu made their first contribution in https://github.com/immich-app/immich/pull/13591
* @chriskoch made their first contribution in https://github.com/immich-app/immich/pull/13578
* @HeyBanditoz made their first contribution in https://github.com/immich-app/immich/pull/12223
* @dvbthien made their first contribution in https://github.com/immich-app/immich/pull/13609
* @darrelhong made their first contribution in https://github.com/immich-app/immich/pull/13600
* @jrstrunk made their first contribution in https://github.com/immich-app/immich/pull/13644
* @evie-lau made their first contribution in https://github.com/immich-app/immich/pull/13622
* @polgarc made their first contribution in https://github.com/immich-app/immich/pull/13640
* @AnimeshKotka made their first contribution in https://github.com/immich-app/immich/pull/13654
* @Pranay-Pandey made their first contribution in https://github.com/immich-app/immich/pull/13677
* @automaton82 made their first contribution in https://github.com/immich-app/immich/pull/13708
* @Sweebah made their first contribution in https://github.com/immich-app/immich/pull/13754

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.118.2...v1.119.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.119.0'>v1.119.0</a>.</em>

**Comment by Morthor** [comment]


**Reply by CrushedAsian255** [reply]
Have you opened a support ticket on the Discord?

**Reply by alextran1502** [reply]
sounds like your file system at UPLOAD_LOCATION is unmounted

**Reply by Morthor** [reply]
Apologies. It was an issue on my side.

**Reply by alextran1502** [reply]
No problem, thanks for reporting!

**Comment by T0ytoy** [comment]
Thank you for the update! As always, it is greatly appreciated. 
I have a question: what is hardware decoding useful for? Currently on my instance, every video I upload is initially transcoded (using hardware acceleration!) into a lighter h264 version and stored, this version is then used when a user try to see a video from the app or webpage.

Would hardware decoding let me disable h264 encoding of all my videos, saving space on disk, by encoding video on-the-fly when a user wants to play a video? That would probably be ideal for me, since I only have 1-3 max concurrent users at all time.

Thank you!

**Reply by alextran1502** [reply]
Hardware transcoding only helps with speeding up the transcoding process

**Reply by T0ytoy** [reply]
Could you explain to me what is the difference between 1.118 and 1.119 regarding VAAPI hardware acceleration of video transcoding, if I already enabled VAAPI hardware transcoding in 1.118? Is the difference that before, it used hardware acceleration on part of the transcoding process (of newly added videos), and now it uses it on the whole transcoding process, making it faster (for the same end result)?

I hope I'm making myself clear enough, thank you for your time ;)

**Reply by mertalev** [reply]
> Is the difference that before, it used hardware acceleration on part of the transcoding process (of newly added videos), and now it uses it on the whole transcoding process, making it faster (for the same end result)?

Exactly this.

**Comment by yu-jingrui** [comment]
Android app still at 1.118.0 in play store?

**Reply by alextran1502** [reply]
Yeah PlayStore is getting very slow with their review process

**Comment by drmcachr** [comment]
Tablets UI still has many blank between photos
![IMG_0127](https://github.com/user-attachments/assets/5c26f59b-67de-4632-8c3c-bf508b32ec99)

**Reply by alextran1502** [reply]
You should open an issue and provide where is this view from, which settings have you used so we can track and fix it

**Reply by Meliox** [reply]
I think it is the same as this one. https://github.com/immich-app/immich/pull/13808 that you fixed yesterday. 
In portrait mode I see this when <= 3 thumbnails and in horizontal mode <= 4 thumbnails on my tab s9 - immich 1.119.1. 
I see this on all pages, main, search, library (don't have albums to confirm). Let me know if you want me to create a ticket should above not fix it
